readthedocs/projects/tasks.py

```diff
@@ -1359,30 +1367,28 @@ def warn(self, msg):
                 doc_name=doc_name,
                 doc_display=titles.get(doc_name, ''),
                 anchor=anchor,
+                commit=commit,
+                build=build,
             )
-            if obj.commit != commit:
-                obj.commit = commit
-                obj.save()
-            if created:
-                created_sphinx_domains.append(obj)
-
-    # Send bulk_post_create signal for bulk indexing to Elasticsearch
-    bulk_post_create.send(sender=SphinxDomain, instance_list=created_sphinx_domains, commit=commit)
-
-    # Delete the SphinxDomain first from previous commit and
-    # send bulk_post_delete signal for bulk removing from Elasticsearch
-    delete_queryset = (
-        SphinxDomain.objects.filter(project=version.project,
-                                    version=version
-                                    ).exclude(commit=commit)
+
+    # Index new SphinxDomain objects to elasticsearch
+    index_new_files(model=SphinxDomain, version=version, build=build)
+
+    # Objects from the db must exists before calling this function,
+    # because the task will query the DB for the objects before deleting
```

Comment:
Is this comment true anymore? I don't think it is.

Reply:
You are right, removing it now